tests/conftest.py

```diff
@@ -0,0 +1,31 @@
+import json
+import os
+import pytest
+import requests_mock
+
+
+TEST_API_FIXTURES_PATH = "tests/fixtures/api"
+JSON_FILE = "_.{}.json"
+
+
+@pytest.fixture(scope="session")
+@requests_mock.Mocker(kw="mock")
+def request_mock(mock=None):
+    def mock_response(request, context):
+        path = request.url.split("/api/v1/")[1].split("?")[0]
+        json_path = os.path.join(
+            TEST_API_FIXTURES_PATH, path, JSON_FILE.format(request.method)
+        )
+        with open(json_path) as json_file:
+            return json.load(json_file)
+
+    # Configure requests to use the local JSON files a response
+    mock.register_uri(requests_mock.ANY, requests_mock.ANY, json=mock_response)
+    return mock
+
+
+@pytest.fixture(scope="class")
```

Comment:
Question : I see you defined the scope of the request mock to "class". I assume this is the reason why you have to explictly call the reset method in the test class initialization function. Any reason why you did not define with a "function" scope which I guess should take care of resetting it automatically for you ? I'm not sure I see any benefit of keeping the same request_mock between two separate tests.

Reply:
Done